server/src/main/java/org/apache/iotdb/db/engine/compaction/io/CompactionTsFileWriter.java

```diff
@@ -0,0 +1,120 @@
+import java.io.Serializable;
+
+public class CompactionTsFileWriter extends TsFileIOWriter {
+  CompactionType type;
+
+  private volatile boolean isWritingAligned = false;
+
+  public CompactionTsFileWriter(
+      File file, boolean enableMemoryControl, long maxMetadataSize, CompactionType type)
+      throws IOException {
+    super(file, enableMemoryControl, maxMetadataSize);
+    this.type = type;
+  }
+
+  public void markStartingWritingAligned() {
+    isWritingAligned = true;
+  }
+
+  public void markEndingWritingAligned() {
+    isWritingAligned = false;
+  }
+
+  public void writeChunk(IChunkWriter chunkWriter) throws IOException {
+    boolean isAligned = chunkWriter instanceof AlignedChunkWriterImpl;
+    long beforeOffset = this.getPos();
+    chunkWriter.writeToFileWriter(this);
+    long writtenDataSize = this.getPos() - beforeOffset;
+    if (writtenDataSize > 0) {
+      CompactionTaskManager.getInstance().getMergeWriteRateLimiter().acquire((int) writtenDataSize);
+    }
+    CompactionMetrics.getInstance()
+        .recordWriteInfo(
+            type,
+            isAligned ? CompactionIoDataType.ALIGNED : CompactionIoDataType.NOT_ALIGNED,
+            writtenDataSize);
+  }
+
+  @Override
+  public void writeChunk(Chunk chunk, ChunkMetadata chunkMetadata) throws IOException {
+    long beforeOffset = this.getPos();
+    super.writeChunk(chunk, chunkMetadata);
+    long writtenDataSize = this.getPos() - beforeOffset;
+    if (writtenDataSize > 0) {
+      CompactionTaskManager.getInstance().getMergeWriteRateLimiter().acquire((int) writtenDataSize);
+    }
+    CompactionMetrics.getInstance()
+        .recordWriteInfo(
+            type,
+            isWritingAligned ? CompactionIoDataType.ALIGNED : CompactionIoDataType.NOT_ALIGNED,
+            writtenDataSize);
+  }
+
+  @Override
+  public void writeEmptyValueChunk(
+      String measurementId,
+      CompressionType compressionType,
+      TSDataType tsDataType,
+      TSEncoding encodingType,
+      Statistics<? extends Serializable> statistics)
+      throws IOException {
+    long beforeOffset = this.getPos();
+    super.writeEmptyValueChunk(
+        measurementId, compressionType, tsDataType, encodingType, statistics);
+    long writtenDataSize = this.getPos() - beforeOffset;
+    CompactionMetrics.getInstance()
+        .recordWriteInfo(type, CompactionIoDataType.ALIGNED, writtenDataSize);
+    if (writtenDataSize > 0) {
+      CompactionTaskManager.getInstance().getMergeWriteRateLimiter().acquire((int) writtenDataSize);
+    }
+  }
+
+  @Override
+  public void endFile() throws IOException {
+    long beforeSize = this.getPos();
+    super.endFile();
+    long writtenDataSize = this.getPos() - beforeSize;
+    if (writtenDataSize > 0) {
+      CompactionTaskManager.getInstance().getMergeWriteRateLimiter().acquire((int) writtenDataSize);
+    }
+    CompactionMetrics.getInstance()
+        .recordWriteInfo(type, CompactionIoDataType.METADATA, writtenDataSize);
+  }
+}
```

Comment:
It seems that there is no record of the amount of data written to the temporary file during the compaction process.